### PR TITLE
We return false if we cannot login with user and password - this allo…

### LINKED ISSUE #25362: Cannot trust servers with Federation app in oC 9.1 RC1 version
### Steps to reproduce

1. Configure two certified oC servers server1 and server2
2. In both servers, modify in /opt/owncloud/config.php the following line
e.g 'overwrite.cli.url' => 'https://server1' and 'https://server2' respectively
3. In admin menu, in Federation section, we write down the url of the other server
4. Back to the CLI, we run the following commands in /opt/owncloud:
```
sudo -u www-data php cron.php (on server1, server2, server1, server2)
sudo -u www-data ./occ dav:sync-system-addressbook (on server1 a server2) 
sudo -u www-data ./occ federation:sync-addressbooks (on server1, server2)
```

### Expected behaviour

After that the address books should be synced and auto complete should work. Trusted server gauge in webUI coloured in green

### Actual behaviour

sudo -u www-data ./occ federation:sync-addressbooks
    1 [->--------------------------]Error while syncing https://docker.oc.solidgear.es:58359 : Unauthorized

Note: It worked fine in oC 9.1.0. pre alpha version
### Server configuration
**Operating system**:
Ubuntu 14.04

**Web server:**
Apache

**Database:**
MySQL

**PHP version:**
5.5.9

**ownCloud version:"9.1.0.8","versionstring":"9.1.0 beta 2","edition":"Enterprise"

**Updated from an older ownCloud or fresh install:**
Fresh

**Are you using external storage, if yes which one:** 
No

**Are you using encryption:** 
No

**Logs**
```
```
### Client configuration

**browser**

Firefox

### PR DESCRIPTION
**Pull request by DeepDiver1975**
…ws other auth plugins to run.

fixes https://github.com/owncloud/core/issues/25362

@PVince81 @davitol

### CODE REVIEW

File: apps/dav/lib/Connector/Sabre/Exception/PasswordLoginForbidden.php [hunk 1]
```diff
@@ -1,54 +0,0 @@
...
-	 * @param Server $server
-	 * @param DOMElement $errorNode
-	 * @return void
-	 */
-	public function serialize(Server $server, DOMElement $errorNode) {
-
-		// set ownCloud namespace
-		$errorNode->setAttribute('xmlns:o', self::NS_OWNCLOUD);
-
-		$error = $errorNode->ownerDocument->createElementNS('o:', 'o:hint', 'password login forbidden');
```
Comment (ChristophWurst) [inline apps/dav/lib/Connector/Sabre/Exception/PasswordLoginForbidden.php]: how will clients detect that they need to use a token instead without this hint?

Comment (DeepDiver1975) [inline apps/dav/lib/Connector/Sabre/Exception/PasswordLoginForbidden.php]: no idea - but this approach is killing auth between servers :cry:

**Comment by DeepDiver1975** [comment]
I have another idea - let me come up with a different approach in the next hours.

File: apps/dav/lib/Connector/Sabre/Exception/PasswordLoginForbidden.php (hunk 1 above)
Comment (PVince81) [inline apps/dav/lib/Connector/Sabre/Exception/PasswordLoginForbidden.php]: Is there a way to change the order of auth plugins to have the one from fed shares to run first and have the one above be the last one ?

Comment (DeepDiver1975) [inline apps/dav/lib/Connector/Sabre/Exception/PasswordLoginForbidden.php]: Exactly what the alternative pr tries to do.

**Pull request closed without merging**